### PR TITLE
Merge spline 0.5.3 into aws-poc

### PR DESCRIPTION
**Pull request by Zejnilovic**
So this is just a merge of branch `feature/735-integrate-spline-0-4-x` into `aws-poc`, I am not sure how well it is going to work but for the purposes of PoC it should be enough.

### CODE REVIEW

File: menas/src/test/resources/application.properties [hunk 1]
```diff
@@ -68,7 +68,9 @@ menas.environment=localhost
 
 menas.spark.master=local[1]
 
-spline.urlTemplate=http://localhost:8080/spline/dataset/lineage/_search?path=%s&application_id=%s
+# Avoiding the http(s) protocol will cause that the same protocol will be used as for Menas itself, avoiding issues
+menas.lineage.readApiUrl=//localhost:8080/spline-gw/consumer
+lineageService=//localhost:8080/spline-ui/app/lineage-overview?executionEventId=%s
```
Comment (benedeki) [inline menas/src/test/resources/application.properties:73]: I am not sure about this configuration 🤔

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala [hunk 2]
```diff
@@ -86,12 +86,12 @@ trait CommonJobExecution {
     dao.authenticate()
     val dataset = dao.getDataset(cmd.datasetName, cmd.datasetVersion)
     val reportVersion = getReportVersion(cmd, dataset)
-    val pathCfg = getPathConfig(cmd, dataset, reportVersion)
+    val pathCfg: PathConfig = getPathConfig(cmd, dataset, reportVersion)
```
Comment (benedeki) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:89]: Not a problem, just curious - why did you add the type declaration? :-)

**Review by benedeki — APPROVED** [review]
In fast review it looks good. Will look at it in more detail on Friday, if still relevant.

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala (hunk 2 above)
Comment (Zejnilovic) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:89]: Good question. Might be a misclick.

File: menas/src/test/resources/application.properties (hunk 1 above)
Comment (Zejnilovic) [reply menas/src/test/resources/application.properties:73]: It shouldn't be there.

**Review by benedeki — APPROVED** [review]

**Pull request merged**